probe-rs/src/flashing/loader.rs

```diff
@@ -21,142 +21,195 @@ use crate::memory::MemoryInterface;
 use crate::session::Session;
 use crate::Target;
 
-/// `FlashLoader` is a struct which manages the flashing of any chunks of data onto any sections of flash.
-///
-/// Use [add_data()](FlashLoader::add_data) to add a chunk of data.
-/// Once you are done adding all your data, use `commit()` to flash the data.
-/// The flash loader will make sure to select the appropriate flash region for the right data chunks.
-/// Region crossing data chunks are allowed as long as the regions are contiguous.
-pub struct FlashLoader {
-    memory_map: Vec<MemoryRegion>,
-    builder: FlashBuilder,
-
-    /// Source of the flash description,
-    /// used for diagnostics.
-    source: TargetDescriptionSource,
+/// Helper trait for object safety.
+pub trait ImageReader: Read + Seek {}
+impl<T> ImageReader for T where T: Read + Seek {}
+
+/// Load and parse a firmware in a particular format, and add it to the flash loader.
```

Comment:
Could you mention here that we also "convert" the firmware? IDF is not an input format.